data/src/main/java/com/microsoft/azure/kusto/data/auth/CloudInfo.java

```diff
@@ -122,41 +118,45 @@ private static CloudInfo fetchImpl(String clusterUrl, @Nullable HttpClient given
         CloudInfo result;
         HttpClient localHttpClient = givenHttpClient == null ? HttpClientFactory.create(null) : givenHttpClient;
         try {
-            HttpGet request = new HttpGet(UriUtils.appendPathToUri(clusterUrl, METADATA_ENDPOINT));
-            request.addHeader(HttpHeaders.ACCEPT_ENCODING, "gzip,deflate");
-            request.addHeader(HttpHeaders.ACCEPT, "application/json");
+            HttpRequest request = new HttpRequest(HttpMethod.GET, UriUtils.appendPathToUri(clusterUrl, METADATA_ENDPOINT));
+            request.setHeader(HttpHeaderName.ACCEPT_ENCODING, "gzip,deflate");
+            request.setHeader(HttpHeaderName.ACCEPT, "application/json");
 
+            // Fixme: Make me not synchronous please
             // trace CloudInfo.httpCall
             HttpResponse response = MonitoredActivity.invoke(
-                    (SupplierOneException<HttpResponse, IOException>) () -> localHttpClient.execute(request),
+                    (SupplierOneException<HttpResponse, IOException>) () -> localHttpClient.sendSync(request, new Context(new Object(), null)),
                     "CloudInfo.httpCall");
             try {
-                int statusCode = response.getStatusLine().getStatusCode();
+                int statusCode = response.getStatusCode();
                 if (statusCode == HttpStatus.SC_OK) {
-                    String content = EntityUtils.toString(response.getEntity());
+                    String content = response.getBodyAsString().block();
                     if (content == null || content.equals("") || content.equals("{}")) {
                         throw new DataServiceException(clusterUrl, "Error in metadata endpoint, received no data", true);
                     }
                     result = parseCloudInfo(content);
                 } else if (statusCode == 404) {
                     result = DEFAULT_CLOUD;
                 } else {
-                    String errorFromResponse = EntityUtils.toString(response.getEntity());
+                    String errorFromResponse = response.getBodyAsString().block();
                     if (errorFromResponse.isEmpty()) {
-                        errorFromResponse = response.getStatusLine().getReasonPhrase();
+                        // Fixme: Missing reason phrase to add to exception. Potentially want to use an enum.
+                        errorFromResponse = "";
                     }
-                    throw new DataServiceException(clusterUrl,
-                            "Error in metadata endpoint, got code: " + statusCode + "\nWith error: " + errorFromResponse,
-                            statusCode != HttpStatus.SC_TOO_MANY_REQUESTS);
+                    throw new DataServiceException(clusterUrl, "Error in metadata endpoint, got code: " + statusCode +
+                            "\nWith error: " + errorFromResponse, statusCode != 429);
```

Comment:
The original const values that were used were from the Apache library but if we want to look into removing Apache HttpCore as well the HttpClient, we'll need to use constants from somewhere else. 

Those same const values from Apache are provided in another 3rd party library that the Kusto library uses (nimbus) but I don't think it would be a good idea to rely on that dependency being around forever. If Nimbus removes Apache or nimbus needs removed from Kusto dependencies we'll run into the same problem again.

So what I've done is create my own HttpStatus class with public static final int values. I put this class under the http package in data.